index.js

```diff
@@ -9,31 +12,113 @@ function ask(questionText) {
 
 // remember the StateMachine lecture
 // https://bootcamp.burlingtoncodeacademy.com/lessons/cs/state-machines
-let states = {
-  'roomOne': { canChangeTo: [ 'roomTwo' ] },
-  'roomTwo': { canChangeTo: [ 'roomThree' ] },
-  'roomThree': { canChangeTo: [ 'roomOne' ] }
+
+let rooms = {     // was given in starter code 
+  'outside':   { 
+    canMoveTo: [ 'foyer' ],    
+    roomMessage: `182 Main St.
+  You are standing on Main Street between Church and South Winooski.
+  There is a door here. A keypad sits on the handle.
+  On the door is a handwritten sign.`, 
+    signMsg: `The sign says "Welcome to Burlington Code Academy! Come on up to
+    the third floor. If the door is locked, use the code 12345."`, 
+    exitKey: `12345` },
+  'foyer':     {
+    canMoveTo: [ `stairway` ] ,
+    roomMessage: `You are in foyer`,
+    signMsg: `This sign says nothing.`,                           
+    exitKey: `exit1` },
+  'stairway':  {
+    canMoveTo: [ `hallway` ],   
+    roomMessage: `You are in stairway`,
+    signMsg: `This sign says nothing.`,                           
+    exitKey: `exit2` },
+  'hallway':   {
+    canMoveTo: [ 'classroom' ], 
+    roomMessage: `You are in hallway`,
+    signMsg: `This sign says nothing.`,                           
+    exitKey: `exit3` },
+  'classroom': {
+    canMoveTo: [ 'hallway' ],   
+    roomMessage: `You are in classroom!`,
+    signMsg: `This sign says "classroom".`,                       
+    exitKey: `exit4` }
 };
+let currentRoom = "outside";
 
-let currentState = "green";
+var player = {
+  currentState: currentRoom, 
+  inventory: []
+}
 
-function enterState(newState) {
-  let validTransitions = states[currentState].canChangeTo;
-  if (validTransitions.includes(newState)) {
-    currentState = newState;
+function enterRoom(newRoom) {
+  let validTransitions = rooms[currentRoom].canMoveTo;
+  if (validTransitions.includes(newRoom)) {
+    currentRoom = newRoom;
   } else {
-    throw 'Invalid state transition attempted - from ' + currentState + ' to ' + newState;
+    throw 'Invalid room transition attempted - from ' + currentRoom + ' to ' + newRoom;
   }
-}
 
 start();
```

Comment:
This line never gets called because it is inside the enterRoom function definition